lib/diego/client.rb

```diff
@@ -168,9 +179,23 @@ def with_request_error_handling(&blk)
       raise RequestError.new(e.message)
     end
 
+    def active_bbs_url
+      uri = URI(base_url)
+      uri.host = "#{active_bbs_id}.#{uri.host}"
+      uri.to_s
+    end
+
+    def active_bbs_id
```

Comment:
Querying the active bbs instance for every call seems rather costly. I would propose to cache the `active_bbs_url` (in some instance field) and clear it in the `rescue` block in `with_request_error_handling`.

Reply:
Yeah that seems reasonable. I'll take a look at that now.

Reply:
Done this. We store the active instance id in a variable and keep reusing that until it fails, then get a new one from locket.